rent/src/lib.rs

```diff
@@ -0,0 +1,1489 @@
+				current_actions += 1;
+				if current_actions >= max_actions {
+					break
+				}
+			}
+
+			// Subscription queue management
+			while let Some(nft_id) = queues.subscription_queue.pop_next(now) {
+				let contract = match Contracts::<T>::get(nft_id) {
+					Some(x) => x,
+					None => continue,
+				};
+				read += 1;
+				let (should_renew, revoker) = match Self::should_renew(&contract, now) {
+					Err(_err) => continue,
+					Ok(x) => x,
+				};
+				if should_renew {
+					match Self::renew_contract(RawOrigin::Root.into(), nft_id) {
+						Err(_err) => continue,
+						Ok(_) => (),
+					}
+					if let Some(expiration_block) =
+						Self::get_expiration_block(&contract.duration, now)
+					{
+						match queues.insert_in_queue(nft_id, &contract.duration, expiration_block) {
+							Err(_err) => continue,
+							Ok(()) => (),
+						}
+					}
+					read += 3;
+					write += 1;
+				} else {
+					let _ = Self::end_contract(RawOrigin::Root.into(), nft_id, revoker);
+					read += 2;
+					write += 2;
+				}
+				current_actions += 1;
+				if current_actions >= max_actions {
+					break
+				}
+			}
+
+			// Available queue management
+			while let Some(nft_id) = queues.available_queue.pop_next(now) {
+				let _ = Self::remove_expired_contract(RawOrigin::Root.into(), nft_id);
+				read += 3;
+				write += 5;
+				current_actions += 1;
+				if current_actions >= max_actions {
+					break
+				}
+			}
+
+			if current_actions > 0 {
+				Queues::<T>::set(queues);
+				write += 1;
+			}
+
+			T::DbWeight::get().reads_writes(read, write)
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create a new rent contract with the provided details.
+		#[pallet::weight((
+			{
+				let s = Queues::<T>::get().available_queue.size();
+				T::WeightInfo::create_contract(s as u32)
+			},
+			DispatchClass::Normal
+		))]
+		pub fn create_contract(
+			origin: OriginFor<T>,
+			nft_id: NFTId,
+			duration: Duration<T::BlockNumber>,
+			acceptance_type: AcceptanceType<AccountList<T::AccountId, T::AccountSizeLimit>>,
+			revocation_type: RevocationType,
+			rent_fee: RentFee<BalanceOf<T>>,
+			renter_cancellation_fee: Option<CancellationFee<BalanceOf<T>>>,
+			rentee_cancellation_fee: Option<CancellationFee<BalanceOf<T>>>,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let mut queues = Queues::<T>::get();
+
+			// Check number of contract
+			ensure!(
+				queues.total_size() + 1 <= T::SimultaneousContractLimit::get(),
+				Error::<T>::MaxSimultaneousContractReached
+			);
+
+			// Check nft data.
+			let mut nft = T::NFTExt::get_nft(nft_id).ok_or(Error::<T>::NFTNotFound)?;
+			ensure!(nft.owner == who, Error::<T>::NotTheNFTOwner);
+			Self::ensure_nft_available(&nft)?;
+
+			// Check contract parameters.
+			Self::check_contract_parameters(
+				nft_id,
+				&duration,
+				&revocation_type,
+				&rent_fee,
+				&renter_cancellation_fee,
+				&rentee_cancellation_fee,
+			)?;
+			// Ensure that rent fee is valid
```

Comment:
This is unnecessary. The comment is repeating the function name. Do this for other cases too.